src/CoreShop/Component/Core/Order/Modifier/CartItemQuantityModifier.php

```diff
@@ -13,25 +13,90 @@
 namespace CoreShop\Component\Core\Order\Modifier;
 
 use CoreShop\Component\Core\Model\CartItemInterface;
+use CoreShop\Component\Core\Model\ProductInterface;
+use CoreShop\Component\Product\Model\ProductUnitDefinitionInterface;
 use CoreShop\Component\StorageList\Model\StorageListItemInterface;
 use CoreShop\Component\StorageList\StorageListItemQuantityModifierInterface;
 use Webmozart\Assert\Assert;
 
 class CartItemQuantityModifier implements StorageListItemQuantityModifierInterface
 {
-    public function modify(StorageListItemInterface $item, int $targetQuantity)
+    /**
+     * @param StorageListItemInterface $item
+     * @param float                    $targetQuantity
+     */
+    public function modify(StorageListItemInterface $item, float $targetQuantity)
     {
         /**
          * @var CartItemInterface $item
          */
         Assert::isInstanceOf($item, CartItemInterface::class);
 
-        $item->setQuantity($targetQuantity);
+        $cleanTargetQuantity = $this->roundQuantity($item, $targetQuantity);
+
+        $item->setQuantity($cleanTargetQuantity);
 
         if ($item->hasUnitDefinition()) {
             $item->setDefaultUnitQuantity($item->getUnitDefinition()->getConversionRate() * $item->getQuantity());
         } else {
             $item->setDefaultUnitQuantity($item->getQuantity());
         }
     }
+
+    /**
+     * @param StorageListItemInterface $item
+     * @param float $targetQuantity
+     *
+     * @return float
+     */
+    public function roundQuantity(StorageListItemInterface $item, float $targetQuantity)
+    {
+        if (!$item instanceof CartItemInterface) {
+            return $targetQuantity;
+        }
+
+        if (!$item->hasUnitDefinition()) {
+            return $targetQuantity;
+        }
+
+        $product = $item->getProduct();
+        if (!$product instanceof ProductInterface) {
+            return $targetQuantity;
+        }
+
+        $scale = $this->getScale($item);
+        if ($scale === null) {
+            return $targetQuantity;
```

Comment:
since it has no scale here, shouldn't we remove possible floating numbers?

Reply:
the return value of the `getScale` method can be `null` or `integer`, no floating numbers are possible here.